src/pages/Dictionary/components/EditMode.jsx

```diff
@@ -1,53 +1,103 @@
-import { Button, Textarea } from '@nextui-org/react';
+import { useRef } from 'react';
+import { Button, Textarea, useDisclosure } from '@nextui-org/react';
+import { toast } from 'sonner';
+import { useNavigate } from 'react-router-dom';
+import getEditEntriesMutation from '../../../utils/getEditEntriesMutation';
+import EditConfirmModal from './EditConfirmModel';
 
 const EditMode = ({ entry, setEditEntryid }) => {
+  const { isOpen, onOpen, onOpenChange } = useDisclosure();
+  const definition = useRef(entry.definition);
+  const example = useRef(entry.example);
+
+  const navigate = useNavigate();
+  const mutation = getEditEntriesMutation(entry.id);
+
   const handleEditCancel = () => {
     setEditEntryid(undefined);
   };
 
-  const handleEditSubmit = (e) => {
+  const handleUpdate = (e) => {
     e.preventDefault();
-    const formData = new FormData(e.target);
-    const definition = formData.get('definition');
-    const example = formData.get('example');
```

Comment:
i preferred using form because when the required field is empty it automatically focuses the empty field and prevent submission but it's fine

Reply:
I tried using form data, but when I introduced the modal the form doesn't not have the e.target any more.

Reply:
I had to switch it to ref. I was trying to get formData to work because I knew you preferred it because you refactored the other code to use it.

Reply:
I couldn't get it to work properly.

Reply:
Unless we take out the modal confirmation, but I think the modal confirmation is good.

Reply:
Let me know if you have suggestions.

Reply:
Let me know if you have suggestions.

Reply:
you can move the modal compoment inside the form tag

Reply:
Hmm. I thought I tried that. Let me try it again.

Reply:
@sonyakim-dev See doesn't work:

<img width="1358" alt="Screenshot 2024-03-12 at 7 54 38 PM" src="https://github.com/rainawan/BruinDictionary/assets/49802389/1a7fa2c0-66ca-4c9c-960e-39424d757000">

Reply:
I might have figured a way stand by.

Reply:
New solution. Use both form data and use ref. Check it out.  You have your form validation you like.

Reply:
ah i you cannot, its fine

Reply:
I put form data back! I'm using it with ref. The validation is there!